src/components/v2/V2Project/V2PayButton/V2PayForm/V2PayForm.tsx

```diff
@@ -0,0 +1,160 @@
+import { ThemeContext } from 'contexts/themeContext'
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+import { isAddress } from 'ethers/lib/utils'
+import { useContext, useState } from 'react'
+
+import { ProjectPreferences } from 'constants/v2/projectPreferences'
+import { StickerSelection } from './StickerSelection'
+
+export interface V2PayFormType {
+  memo?: string
+  beneficiary?: string
+  stickerUrls?: string[]
+  uploadedImage?: string
+  preferClaimed?: boolean
+}
+
+export const V2PayForm = ({ form }: { form: FormInstance<V2PayFormType> }) => {
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+
+  const { tokenAddress } = useContext(V2ProjectContext)
+
+  const [customBeneficiaryEnabled, setCustomBeneficiaryEnabled] =
+    useState<boolean>(false)
+
+  const [attachStickerModalVisible, setAttachStickerModalVisible] =
+    useState<boolean>(false)
+
+  const hasIssuedTokens = tokenAddress !== constants.AddressZero
+
+  const stickerUrls = useWatch('stickerUrls', form)
+
+  const canAddMoreStickers =
+    (stickerUrls ?? []).length < ProjectPreferences.STICKER_MAX
+
+  return (
+    <>
+      <Form form={form} layout="vertical">
+        <Form.Item
+          label={t`Memo (optional)`}
+          className={'antd-no-number-handler'}
+          extra={t`Add an on-chain memo to this payment.`}
+        >
+          <Form.Item noStyle name="memo">
+            <Input.TextArea
+              placeholder={t`WAGMI!`}
+              maxLength={256}
+              showCount
+              autoSize
+            />
+          </Form.Item>
+          <div
+            style={{
+              fontSize: '.8rem',
+              position: 'absolute',
+              right: 7,
+              top: 7,
+            }}
+          >
+            {canAddMoreStickers ? (
+              <Tooltip title={t`Attach a sticker`}>
+                <SmileOutlined
+                  style={{ color: colors.text.secondary }}
+                  onClick={() => {
+                    setAttachStickerModalVisible(true)
+                  }}
+                />
+              </Tooltip>
+            ) : (
+              <SmileOutlined
+                style={{ color: colors.text.disabled, cursor: 'not-allowed' }}
+              />
+            )}
+          </div>
+          <Form.Item name="stickerUrls">
+            <StickerSelection />
+          </Form.Item>
+        </Form.Item>
+        <Form.Item name="uploadedImage">
+          <FormImageUploader text={t`Add image`} />
+        </Form.Item>
+        <Form.Item extra={t`Mint tokens to a custom address.`}>
+          <span style={{ color: colors.text.primary, fontWeight: 500 }}>
+            <Trans>Custom token beneficiary</Trans>
+          </span>
+          <Switch
+            checked={customBeneficiaryEnabled}
+            onChange={setCustomBeneficiaryEnabled}
+            style={{ marginLeft: 10 }}
+          />
```

Comment:
@wraeth-eth sorry one more thing. Can we flip the switch and the text the other way around?

```
<switch> Custom token beneficiary
```

Reply:
<img width="623" alt="image" src="https://user-images.githubusercontent.com/104132113/177685344-cc3a60ed-f4bb-4563-980e-a7dd6c3ee544.png">

Reply:
<img width="627" alt="image" src="https://user-images.githubusercontent.com/104132113/177685773-8cc7958b-e230-4546-91d0-6398bb9bd0dd.png">
dropping the margin to align better wdyt

Reply:
Yep nailed it